### ISSUE TITLE
jsx approved image alt text

### ISSUE THREAD

**Issue by glitteringkatie** [mdx-js/mdx#1578]
#### tl;dr quotes in image alt text break babel because they don't get appropriately turned into jsx expressions ([since jsx strings don't support `\"` in them](https://github.com/evanw/esbuild/issues/959))

Hi! I found that if someone uses quotation marks in their image alt (`![something really "cool"](https://www.humanesociety.org/sites/default/files/styles/1240x698/public/2020-07/kitten-510651.jpg)`), the outputted jsx is `<img src="https://www.humanesociety.org/sites/default/files/styles/1240x698/public/2020-07/kitten-510651.jpg" alt="something really \"cool\"" parentName="p" />` Which causes babel to blow up because having escaped " marks in a JSX string is not supported turns out! [This comment](https://github.com/evanw/esbuild/issues/959#issuecomment-797714085) really helped me figure that out. So! I think the solution is to make sure that if an image alt has quotes, those should appropriately be wrapped in a jsx expression `{}`.

Disclaimers: I _think_ this should be on MDX since remark is agnostic to jsx. However, I noticed this wasn't an issue with the link text (Like `[link text "here"](blah.com)`) and that gets handled just fine (aka gets wrapped in appropriate curly braces)--unsure if that was because of something in this project or in remark.
### Your environment

*   **OS**: MacOS
*   **Packages**: 
    * "@mdx-js/mdx@2.0.0-next.9"
    * "@mdx-js/react@2.0.0-next.9"
    * "@mdx-js/util@2.0.0-next.8"
*   **Env**
    * "@babel/core@7.13.14"
    * "@babel/preset-env@7.13.12"
    * "@babel/preset-react@7.13.13"

### Steps to reproduce

1. have an mdx file with `![something really "cool"](https://www.humanesociety.org/sites/default/files/styles/1240x698/public/2020-07/kitten-510651.jpg)` in it somewhere
2. parse that file with mdx
3. put that parsed content through babel (`transformAsync` from babel core with presetReact & presetEnv)

🎉 BONUS POINTS for creating a [minimal reproduction](https://stackoverflow.com/help/mcve) and uploading it to GitHub.  This will get you the fastest support.  🎉

### Expected behaviour

Babel is happy and outputs functioning code!

### Actual behaviour

Babel is sad and says "SyntaxError: unknown: Expecting Unicode escape sequence \uXXXX"

### Work around

For now I'm making a quick remark plugin that replaces double quotes with `&quot;`! This definitely feels more like a bandaid than a real solution but want to share in case anyone hits this issue and needs a quick fix

```js
import { Node } from 'unist';
import visit from 'unist-util-visit';

export function remarkImageAltQuotes() {
  return function transformer(tree: Node): void {
    visit(tree, ['image'], node => {
      const alt = node.alt as string;
      node.alt = alt.replace(/"/g, '&quot;');
    });
  };
}
```

**Comment by ChristianMurphy** [comment]
Thanks for the detailed report @glitteringkatie! :bow: 
And for providing an example, putting the example in a sandbox I'm seeing the same thing in MDX runtime https://codesandbox.io/s/clever-river-dko1k?file=/src/App.js

Though it does appear to be fixed already in XDM https://codesandbox.io/s/xdm-create-react-app-starter-forked-z8esj?file=/src/App.js

The fix in XDM appears to be https://github.com/wooorm/xdm/blob/fa8fb4918099ff085de45d03d01aa2f952228a8d/lib/plugin/recma-stringify.js#L103-L111
Which is not currently in place at https://github.com/mdx-js/mdx/blob/907bd26d83287180c7155c0fb65671b7cb55c98d/packages/mdx/estree-to-js.js#L31-L34

**Comment by glitteringkatie** [comment]
Ooh, awesome @ChristianMurphy! I was hoping it would be something like that. I am planning to get a fix in PR today or Monday.
